### PR TITLE
Lit poc

### PR DESCRIPTION
**Pull request by Gabez0r**
This seems to break in Web UI because of: `TypeError: host.addController is not a function`. This happens because of conflicting versions of lit amongst transitive dependencies. This points to the **first limitation**: it requires all dependencies (including transitive ones) to have compatible version of lit.

A **second limitation** was spotted: with the most recent version of lit-element, it seems that using `TemplateResult`s inside a `<template>` tag won't work, so it requires us to rely on the unsafeHTML directive.

A **third limitation** is that storybook is broken, mainly because the @storybook/polymer that we use relies on old lit code. The package does not seem to exist anymore, so we might have to upgrade storybook too to have it working.

### CODE REVIEW

**Comment by Gabez0r** [comment]
This seems to break in Web UI because of: TypeError: host.addController is not a function.

**Comment by Gabez0r** [comment]
Requirements:
- Upgrade to [lit](https://lit.dev/) v2.* (see ba55383ec02d2a394815119ec2bb2fdb1512bcbb).
- Disable polymer lint, because it breaks (788c018a3af2dcc166f3d091d3206e39e2be7d82) - it doesn't do much at this point.
- Update to @open-wc/testing-helpers v2.* (87233cca142680808aa19a933b68f4b0aac0a72f).
- Update vaadin-date-picker, to prevent conflicts with outdated lit versions (see 206f93b262aee6e19bb4c109795ec07a5aff11dd).

**Comment by Gabez0r** [comment]
We chose a `nuxeo-date` to be migrated to lit. It is a simple element but relies heavily on polymer behaviors. Migrating the element was pretty straight forward (see 80af6d557673cc83271fceefcdd03528569a6356), by following lit [documentation](https://lit.dev/docs/). The trickiest part was indeed the reliance on polymer behaviors. The approach depicted in this PR to convert the element is as follows:
1. Extract behavior logic into a ESM module, which exports functions and objects that can be shared. This logic is then used by the behavior implementation (see 0ce37b1983c4e231c6130ebf5d1d4d0f2a63a619 and eaf0d6ae783a57629bc3610e16ccf375f809c115).
2. Create a [reactive controller](https://lit.dev/docs/composition/controllers/) per behaviors that plugs into lifecycle callbacks (such as the `I18nBehavior`), relying on the extracted ESM module for implementation (see ba73317169f43ce9f0ab87a788f932b9d2e23236). The converted element should then add this controller.
3. Behaviors that don't rely on lifecycle callbacks can usually be replaced by direct calls to the methods exported by the shared ESM module.

**Comment by Gabez0r** [comment]
Note that 6b4b4d3c461418a602414a784556b5071a888d0c was added just to allow the story book build to go through as the new testing helpers bring ?? and ?. which need to be transpilled. However, this does not actually make storybook work properly.

File: ui/controllers/nuxeo-i18n-controller.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import '../nuxeo-i18n.js';
+
+/**
+ * Ideally, `I18nController` would implement the `ReactiveController` interface.
```
Comment (richardsd) [inline ui/controllers/nuxeo-i18n-controller.js:20]: You mean [this](https://lit.dev/docs/composition/controllers/), right? Why didn't you implement it?

**Review by richardsd — COMMENTED** [review]
It looks fine by me, don't see anything in particular that is not pretty straightforward to understand.

File: ui/controllers/nuxeo-i18n-controller.js (hunk 1 above)
Comment (Gabez0r) [reply ui/controllers/nuxeo-i18n-controller.js:20]: Sorry, I meant explicitly implement the interface, which can only be done in typescript. I did implement the methods, though. Should I amend the comment?

**Pull request closed without merging**